lib/resty/upstream/healthcheck.lua

```diff
@@ -150,6 +154,7 @@ end
 
 local function peer_ok(ctx, is_backup, id, peer)
     debug("peer ", peer.name, " was checked to be ok")
+    peer_status[peer.name] = "ok"
```

Comment:
I don't think this will work as you expect. The peer is not always marked as `up` by this function. If the peer is marked as up by `upgrade_peers_version` (which happens for on every worker that did not actually run the health check request), the `peer_status` table won't be updated.